slackbridge/bots.py

```diff
@@ -149,6 +151,30 @@ def nickserv_auth(self):
                 self.target_group_nick,
                 self.nickserv_password,
             ))
+    def privmsg(self, user, channel, message):
+        """
+        Set to handle if channel is own name (private chat)
+        """
+
+        nick = utils.nick_from_irc_user(user)
+        
+        if channel == self.nickname:
+            if not self.im_id:
+                self.im_id = self.sc.api_call(
+                        'im.open',
+                        user=self.user_id,
+                        return_im=True
+                        )['channel']['id']
+
+            log.msg(self.sc.api_call(
+                'chat.postMessage',
+                channel=self.im_id,
+                text=utils.format_slack_message(nick + ": " + message, IRCBot.users),
+                as_user=False,
+                username=nick,
+                icon_url=utils.user_to_gravatar(nick),
+            ))
```

Comment:
Instead of doing this whole message, you should be able to use the [`BridgeBot` `post_to_slack` method](https://github.com/ocf/slackbridge/blob/fcb8f85d082aaf58d97fcd964872e0ebbd470bc1/slackbridge/bots.py#L67-L78) and pass in the correct channel and it'll even fetch the nickname for you, making the `nick = utils.nick_from_irc_user(user)` line redundant. I'd probably put that method somewhere in utils so that both the `BridgeBot` and this `UserBot` can use it without having to reach into a different class or something hacky to do it.

Reply:
I'd rather put post_to_slack in the super IRCBot class since this method depends on IRCBot.users and the slack client object being available (otherwise we'd need to pass those around too).